src/hotspot/share/oops/methodCounters.hpp

```diff
@@ -52,17 +52,18 @@ class MethodCounters : public Metadata {
 
   MethodCounters(const methodHandle& mh);
  public:
-  virtual bool is_methodCounters() const { return true; }
-
   static MethodCounters* allocate_no_exception(const methodHandle& mh);
   static MethodCounters* allocate_with_exception(const methodHandle& mh, TRAPS);
 
+  DEBUG_ONLY(bool on_stack() { return false; })
   void deallocate_contents(ClassLoaderData* loader_data) {}
 
+  void metaspace_pointers_do(MetaspaceClosure* it) { return; }
```

Comment:
Does it change the behavior if we remove this along with
` it->push(&_method_counters);`
in Method::metaspace_pointers_do?

Reply:
Yes it's a metaspace pointer.  It's zeroed and not saved in the archive but it still needs to call this function in case it's not zeroed and someday called.